src/impl/load.c

```diff
@@ -0,0 +1,289 @@
+    if (!file) {
+        return NULL;  // Not a valid local file - not our job
+    }
+    fclose(file);
+    
+    // Read file content directly
+    char *gcg_content = get_string_from_file(identifier, error_stack);
+
+    if (!gcg_content || is_string_empty_or_whitespace(gcg_content) ||
+        strstr(gcg_content, "#player1") == NULL) {
+        free(gcg_content);
+        error_stack_push(error_stack, ERROR_STATUS_FILEPATH_FILE_NOT_FOUND,
+                         get_formatted_string("Local file does not contain valid GCG content: %s", identifier));
+        return NULL;
+    }
+
+    return gcg_content;
+}
+
+char *get_gcg_string(const DownloadGCGOptions *options, ErrorStack *error_stack) {
+  
+    if (!options || !options->source_identifier) {
+        error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                         string_duplicate("Invalid options or missing source_identifier"));
+        return NULL;
+    }
+    const char *identifier = options->source_identifier;
+    
+    // Try Cross-tables first 
+    char *gcg_string = get_xt_gcg_string(identifier, error_stack);
+    if (!error_stack_is_empty(error_stack)) {
+        return NULL;  // Fatal error occurred
+    }
+    if (gcg_string) {
+        return gcg_string;
+    }
+
+    // Try Woogles
+    gcg_string = get_woogles_gcg_string(identifier, error_stack);
+    if (!error_stack_is_empty(error_stack)) {
+        return NULL;  // Fatal error occurred
+    }
+    if (gcg_string) {
+        return gcg_string;
+    }
+
+    // Try local file
+    gcg_string = get_local_gcg_string(identifier, error_stack);
+    if (!error_stack_is_empty(error_stack)) {
+        return NULL;  // Fatal error occurred
+    }
+    if (gcg_string) {
+        return gcg_string;
+    }
+
+    // If we get here, nothing worked
+    error_stack_push(error_stack, ERROR_STATUS_FILEPATH_FILE_NOT_FOUND,
+                     get_formatted_string("Could not load GCG from any source: %s", 
+                                          options->source_identifier));
+    return NULL;
+}
+
+void download_gcg(const DownloadGCGOptions *options, GameHistory *game_history, ErrorStack *error_stack) {
+    if (!options || !game_history) {
+        error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                         string_duplicate("Invalid arguments to download_gcg"));
+        return;
+    }
+
+    // Get the GCG content from any available source
+    char *gcg_content = get_gcg_string(options, error_stack);
+    if (!gcg_content) {
+        return;  // Error already pushed to stack by get_gcg_string
+    }
+
+    // Validate config is provided for parsing
+    Config *config_to_use = options->config;
+    if (!config_to_use) {
+        error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                         string_duplicate("GCG content downloaded successfully but parsing requires a valid config"));
+        free(gcg_content);
+        return;
+    }
```

Comment:
I think it's fine to delete this and assume the Config is always in the options.